File: Ejercicio 13 c++/13.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main() {
    ifstream archivo("productos.txt");
    string nombre;
    double precio;

    if (!archivo) {
        cout << "No se pudo abrir el archivo." << endl;
        return 1;
    }

    while (archivo >> nombre >> precio) {
        if (precio > 999) {
            double precioFinal = precio * 0.8;
            cout << nombre << " | Precio original: " << precio 
                 << " | Precio con descuento: " << precioFinal << endl;
        } else {
            cout << nombre << " | Precio: " << precio 
                 << " | Sin descuento" << endl;
        }
    }

    archivo.close();
    return 0;
}

File: Ejercicio 13 py/13.py
def productos_descuento(nombre_archivo):
    try:
        with open(nombre_archivo, "r") as f:
            for linea in f:
                partes = linea.split()
                if len(partes) == 2:
                    nombre = partes[0]
                    precio = float(partes[1])
                    if precio > 999:
                        precioFinal = precio * 0.8
                        print(f"{nombre} | Precio original: {precio} | Precio con descuento: {precioFinal}")
                    else:
                        print(f"{nombre} | Precio: {precio} | Sin descuento")
    except FileNotFoundError:
        print("No se pudo abrir el archivo.")
        
productos_descuento("productos.txt")
